core/che-core-api-core/src/main/java/org/eclipse/che/api/core/util/ProcessUtil.java

```diff
@@ -68,6 +70,55 @@ public static void process(Process p, LineConsumer stdout) throws IOException {
     }
 
     /**
+     * Start the process, writing the stdout and stderr to {@code outputConsumer} and terminate process by {@code timeout}.<br>
+     *
+     * @param commandLine
+     *         arguments of process command
+     * @param timeout
+     *         timeout for process. If process duration > {@code timeout} then kill process and throw {@link TimeoutException}.
+     * @param timeUnit
+     *         timeUnit of the {@code timeout}.
+     * @param outputConsumer
+     *         a consumer where stdout and stderr will be redirected
+     * @return the started process
+     * @throws InterruptedException
+     *         in case terminate process
+     * @throws IOException
+     *         in case I/O error
+     * @throws TimeoutException
+     *         if process gets more time then defined by {@code timeout}
+     */
+    public static Process execute(String[] commandLine, int timeout, TimeUnit timeUnit, LineConsumer outputConsumer)
+            throws TimeoutException, IOException, InterruptedException {
+        ProcessBuilder pb = new ProcessBuilder(commandLine).redirectErrorStream(true);
+
+        // process will be stopped after timeout
+        Watchdog watcher = new Watchdog(timeout, timeUnit);
```

Comment:
Please use name for watchdog thread - it will help understand thread dump in case of troubles with the server.